server/routes/api/v1/allergies/index.js

```diff
@@ -0,0 +1,41 @@
+import { Role } from '../../../../models/user.js';
+
+export default async function (fastify) {
+  fastify.get(
+    '',
+    {
+      schema: {
+        querystring: {
+          allergy: { type: 'string' },
+        },
+      },
+      onRequest: fastify.requireUser([Role.ADMIN, Role.STAFF, Role.VOLUNTEER]),
+    },
+    async (request, reply) => {
+      const allergy = request.query.allergy.trim();
+
+      // prevent empty string from returning all allergies
+      if (!allergy.length) {
+        return reply.send({ message: 'No query provided' });
+      }
```

Comment:
Since this causes a different response structure, should a different behavior be used when no query parameter is provided?